packages/teams-js/src/public/teamsAPIs.ts

```diff
@@ -36,7 +36,11 @@ export namespace teamsCore {
    * default print handler
    */
   export function print(): void {
-    window.print();
+    if (typeof window !== 'undefined') {
```

Comment:
>typeof window !== 'undefined' [](http://example.com/codeflow?start=8&length=29)

Shouldn't this use your `inServerSideRenderingEnvironment` function?

Reply:
Since the point of the `inServerSideRenderingEnvironment` function was to be more semantically meaningful, I was actually thinking in this instance, since the `print` function prints the window, it would be more meaningful and obvious to just directly reference window in the check. I can also go change it to the `inServerSideRenderingEnvironment` function if you strongly disagree with my reasoning or feel the consistency would be better.

Reply:
Hmm, well, I agree that since this function is about printing the `window` then directly checking that the `window` object is available makes that connection clear. I think the readability concern I would have for other developers would be if they came to look at this and thought "That's weird, why would the window object ever be undefined?" I guess I could go either way. 🤷‍♂️

Reply:
I decided to leave the direct check of `window` and supplemented it with a much more descriptive error message that should give other developers a better understanding of why the check is there, and how and answer the "That's weird, why would the window object ever be undefined?" question.